### ISSUE TITLE
spurious syntax error for zsh string interpolations

### ISSUE THREAD

**Issue by mcandre** [koalaman/shellcheck#3431]
#### For bugs with existing features

- Rule Id (if any, e.g. SC1000): SC2296
- My shellcheck version (`shellcheck --version` or "online"): 0.11.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```zsh
_mage() {
    local tasks=( ${(f)"$(mage -l | awk '{ sub(/\*$/, "", $1); if ($1 != "" && $1 != "Targets:") print $1 }')"} )
    _arguments -C '1:task:('"${tasks[*]}"')'
}
compdef _mage mage
```

#### Here's what shellcheck currently says:

```txt
In src/.zshrc.d/autocompletions.zsh line 36:
        local tasks=( ${(f)"$(mage -l | awk '{ sub(/\*$/, "", $1); if ($1 != "" && $1 != "Targets:") print $1 }')"} )
                        ^-^ SC2296 (error): Parameter expansions can't start with (. Double check syntax.
```

#### Here's what I wanted or expected to see:

Nothing. zsh extends POSIX sh with many features, including `(f)` string interpolation syntax to populate arrays from newline delimited strings.

**Comment by brother** [comment]
zsh is not supported (yet?), see #809
